bot/cogs/ext/commands/anime.py

```diff
@@ -47,6 +47,26 @@
     "happy", "wink", "poke", "dance", "cringe"
 ]
 
+class WaifuCommandPaginator(ListPageSource):
+    def __init__(self, data: List[Union[WaifuSearchResult, Waifu]]):
+        super().__init__(data, per_page=1)
+        self.embed = create_embed()
+
+    async def format_page(self, menu: PidroidPages, waifu: Union[Waifu, WaifuSearchResult]):
+        self.embed.clear_fields()
+        if isinstance(waifu, WaifuSearchResult):
+            waifu = await waifu.fetch_waifu()
+        self.embed.title = waifu.name
+        if waifu.is_nsfw and not menu.ctx.channel.is_nsfw():
+            self.embed.set_image(url="")
+            self.embed.description = 'this waifu is nsfw'
```

Comment:
Also, consider changing the following text to something like
```suggestion
            self.embed.description = 'This waifu is tagged as NSFW. In order to view the waifu, please use the command in an age-restricted channel.'
```

Reply:
considered it